### PR TITLE
feature/AKD-09: Added AWS Redshift Nodes Count plugin, test cases

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

**Review by dipsubha06 — APPROVED** [review]

File: plugins/aws/redshift/redshiftNodesCount.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    more_info: 'The number of provisioned Amazon Redshift cluster nodes must be less than the provided nodes limit to avoid reaching the limit and exceeding the set budget.',
+    link: 'https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-clusters.html#working-with-clusters-overview',
+    recommended_action: 'Remove Redshift clusters over defined limit',
+    apis: ['Redshift:describeClusters'],
+    settings: {
+        redshift_nodes_count: {
+            name: 'Amazon Redshift Nodes Count',
+            description: 'Maximum Amazon Redshift nodes count per region',
+            regex: '^.*$',
+            default: 'false'
```
Comment (giorod3) [inline plugins/aws/redshift/redshiftNodesCount.js:17]: by making the default false, it makes it seem like a boolean field. this seems like an integer so it should be 0

Comment (AkhtarAmir) [reply plugins/aws/redshift/redshiftNodesCount.js]: Right. And I think as this is denoting max limit, so it should be 0 as well. So I have set it to 5 by default. Let me know if you want me to change it

**Review by giorod3 — APPROVED** [review]

File: plugins/aws/redshift/redshiftNodesCount.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    more_info: 'The number of provisioned Amazon Redshift cluster nodes must be less than the provided nodes limit to avoid reaching the limit and exceeding the set budget.',
+    link: 'https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-clusters.html#working-with-clusters-overview',
+    recommended_action: 'Remove Redshift clusters over defined limit',
+    apis: ['Redshift:describeClusters'],
+    settings: {
+        redshift_nodes_count: {
+            name: 'Amazon Redshift Nodes Count',
+            description: 'Maximum Amazon Redshift nodes count per region',
+            regex: '^[0-9]{0,3}',
+            default: 5
```
Comment (giorod3) [inline plugins/aws/redshift/redshiftNodesCount.js:17]:
```suggestion
            default: '5'
```

**Review by giorod3 — CHANGES_REQUESTED** [review]
make per region and set default to be 100. also string to int

File: plugins/aws/redshift/redshiftNodesCount.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    description: 'Ensures that each AWS region has not reached the limit set for the number of Redshift cluster nodes.',
+    more_info: 'The number of provisioned Amazon Redshift cluster nodes must be less than the provided nodes limit to avoid reaching the limit and exceeding the set budget.',
+    link: 'https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-clusters.html#working-with-clusters-overview',
+    recommended_action: 'Remove Redshift clusters over defined limit',
+    apis: ['Redshift:describeClusters'],
+    settings: {
+        redshift_nodes_count: {
+            name: 'Amazon Redshift Nodes Count',
+            description: 'Maximum Amazon Redshift nodes count per region',
+            regex: '^[0-9]{0,3}',
```
Comment (giorod3) [inline plugins/aws/redshift/redshiftNodesCount.js:16]:
```suggestion
            regex: '^[0-9]{0,4}',
```

File: plugins/aws/redshift/redshiftNodesCount.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    description: 'Ensures that each AWS region has not reached the limit set for the number of Redshift cluster nodes.',
+    more_info: 'The number of provisioned Amazon Redshift cluster nodes must be less than the provided nodes limit to avoid reaching the limit and exceeding the set budget.',
+    link: 'https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-clusters.html#working-with-clusters-overview',
+    recommended_action: 'Remove Redshift clusters over defined limit',
+    apis: ['Redshift:describeClusters'],
+    settings: {
+        redshift_nodes_count: {
+            name: 'Amazon Redshift Nodes Count',
+            description: 'Maximum Amazon Redshift nodes count per region',
+            regex: '^[0-9]{0,4}',
```
Comment (giorod3) [inline plugins/aws/redshift/redshiftNodesCount.js:16]:
```suggestion
            regex: '^[0-9]{1,4}',
```

**Review by giorod3 — APPROVED** [review]

**Pull request merged**